### DISCUSSION TITLE
More commands examples

### DISCUSSION THREAD

**Discussion by JeBenzon** [ytausch/serverless-discord-bot#33] [Q&A] [answered]
Maybe its just me,
but I got it all up and running (at least I think I do)

But idk how to verify if its working.

What do I do to use the 'helloCommand' in discord?
my bot does not react on /hello or !hello. (maybe this is a indication that I did something wrong, but I got no idea)

**Comment by ytausch** [comment, marked as answer]
Hi,

first, please make sure to set your guild id in `src/commands_layer/nodejs/commands/helloCommand.js`. The application command should then be pushed to Discord after you deployed the SAM application.

To execute the 'helloCommand' command, start typing `/hello` in the text channel of a Discord guild where your bot is authorized. You should then see a suggestion popup like this:

![Screenshot: "Commands with /hello": "/hello"](https://user-images.githubusercontent.com/25906270/150650779-8df12a93-98f3-4822-acae-0fd4315adc4f.png)

After you hit enter, the command should be executed.

If you do not get a suggestion popup, double check that you [authorized your application](https://discord.com/developers/docs/interactions/application-commands#authorizing-your-application) and that your commands were indeed deployed correctly. To do this, open the logs of the `CreateCommandsFunction` lambda by opening the **AWS CloudWatch** console and then navigating to _Logs -> Log group -> /aws/lambda/serverless-discord-bot-CreateCommandsFunction-xxxxxxxxxxx_.

Your log events should contain roughly these lines:

    INFO	[class HelloCommand extends SlashCommand]
    INFO	Registered command xxxxxxxxxxxxx:hello.
    INFO	Found guild command "hello" (xxxxxxxxxxxxx, guild: xxxxxxxxxx)
    INFO	Request: [{
         "id": "xxxxxxxxxxxxxxxxxxxxxx",
        "name": "hello",
        "description": "Says hello to you.",
        "default_permission": true
    }]
    INFO	Finished syncing commands
    INFO	RESPONSE: {
        "Status": "SUCCESS",
        "PhysicalResourceId": "SlashCommands:xxxxxxxxxxxx",
        "StackId": "arn:aws:cloudformation:xxxxxxxxxxxxx",
        "RequestId": "xxxxxxxxxxxx",
        "LogicalResourceId": "CreateCommandsInvoker"
    }


For more information about Discord application commands, make sure to check out the [official documentation on application commands](https://discord.com/developers/docs/interactions/application-commands#application-commands), and the [slash-create docs](https://slash-create.js.org/#/docs). There you will also find more [command examples](https://slash-create.js.org/#/docs/main/latest/examples/command).

**Reply by JeBenzon** [reply]
Thank you for this in depth answer.
I put in my Guild id and now it works!!

Now I just need to figure out how to make cool commands.

**Reply by ytausch** [reply]
Have fun :)
